packages/vite/src/node/preview.ts

```diff
@@ -276,7 +276,14 @@ export async function preview(
   )
 
   if (options.open) {
-    const url = server.resolvedUrls.local[0] ?? server.resolvedUrls.network[0]
+    const host = options.host
+    const url =
+      typeof host === 'string'
+        ? [
+            ...(server.resolvedUrls?.local ?? []),
+            ...(server.resolvedUrls?.network ?? []),
+          ].find((url) => url.includes(host))
+        : (server.resolvedUrls.local[0] ?? server.resolvedUrls.network[0])
```

Comment:
Would you extract this as a function and put it here?
https://github.com/vitejs/vite/blob/0a0c50a7ed38017469ed6dcec941c2d8d0efd0d0/packages/vite/src/node/utils.ts#L1100
so that we use the same function in both serve and preview.